MoreLinq/Aggregate.g.tt

```diff
@@ -0,0 +1,201 @@
+            Arguments = args.Take(a.Count)
+                            .Select(aa => new { aa.Number, Num = aa.Count, aa.Ordinal })
+                            .ToList(),
+        };
+
+    foreach (var o in overloads)
+    {
+#>
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Applies <#= o.Arity #> accumulators sequentially in a single pass over a
+        /// sequence.
+        /// </summary>
+        /// <typeparam name="T">The type of elements in <paramref name="source"/>.</typeparam>
+<#      foreach (var arg in o.Arguments) { #>
+        /// <typeparam name="TAccumulate<#= arg.Number #>">The type of <#= arg.Ordinal #> accumulator value.</typeparam>
+<#      } #>
+        /// <typeparam name="TResult">The type of the accumulated result.</typeparam>
+        /// <param name="source">The source sequence</param>
+<#      foreach (var arg in o.Arguments) { #>
+        /// <param name="seed<#= arg.Number #>">The seed value for the <#= arg.Ordinal #> accumulator.</param>
+        /// <param name="accumulator<#= arg.Number #>">The <#= arg.Ordinal #> accumulator.</param>
+<#      } #>
+        /// <param name="resultSelector">
+        /// A function that projects a single result given the result of each
+        /// accumulator.</param>
+        /// <returns>The value returned by <paramref name="resultSelector"/>.</returns>
+        /// <remarks>
+        /// This operator executes immediately.
+        /// </remarks>
+
+        public static TResult Aggregate<T, <#= string.Join(", ", from x in o.Arguments select "TAccumulate" + x.Number) #>, TResult>(
+            this IEnumerable<T> source,
+<#      foreach (var arg in o.Arguments) { #>
+            TAccumulate<#= arg.Number #> seed<#= arg.Number #>, Func<TAccumulate<#= arg.Number #>, T, TAccumulate<#= arg.Number #>> accumulator<#= arg.Number #>,
+<#      } #>
+            Func<<#= string.Join(", ", from x in o.Arguments select "TAccumulate" + x.Number) #>, TResult> resultSelector)
+        {
+            if (source == null) throw new ArgumentNullException(nameof(source));
+<#      foreach (var arg in o.Arguments) { #>
+            if (accumulator<#= arg.Number #> == null) throw new ArgumentNullException(nameof(accumulator<#= arg.Number #>));
+<#      } #>
+            if (resultSelector == null) throw new ArgumentNullException(nameof(resultSelector));
+
+<#      foreach (var arg in o.Arguments) { #>
+            var a<#= arg.Number #> = seed<#= arg.Number #>;
+<#      } #>
+
+            foreach (var item in source)
+            {
+<#      foreach (var arg in o.Arguments) { #>
+                a<#= arg.Number #> = accumulator<#= arg.Number #>(a<#= arg.Number #>, item);
+<#      } #>
+            }
+
+            return resultSelector(<#= string.Join(", ", from x in o.Arguments select "a" + x.Number) #>);
+        }
+    }
+}
+
+namespace MoreLinq.Experimental
```

Comment:
Yes, speaking strictly, but since _should_ doesn't mean _must_, 😉 I would rather leave it here in order to avoid maintaining duplicate templates. I am hoping the design based on observables won't live in the experimental namespace for too many versions.